### DISCUSSION TITLE
Carousel with v-for

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [quasarframework/quasar#12282] [General - Components / Directives / etc]
I know that the topic has been covered before but I'm still facing the issue, here is my code :

```
<template>
    <q-carousel
      class="carousel"
      animated
      v-model="slide"
      arrows
      navigation
      infinite
    >
      <q-carousel-slide
        v-for="picture in spotInfo.pictures"
        :key="picture.id"
        class="spot-picture"
        :name="picture.id"
        :img-src="cloudFrontUrl + '/' + picture.name"
      />
    </q-carousel>
</template>
<script>
import { ref } from "vue";
setup() {
     return {
       slide: ref(1),
     };
   },
   data() {
     return {
       cloudFrontUrl: "...",
       spotInfo: {...},
      };
   }
</script>
```

I have no idea why but the images are not displayed and there is no http request made for them. Yes spotInfo.pictures is defined as well as picture.id. I think the problem comes from the `name` prop but idk why

**Comment by MilosPaunovic** [comment]
Have you tried to set `img-url` using string literals:
```js
:img-src="`${cloudFrontUrl}/${picture.name}`"
```

**Comment by FrenchGithubUser** [comment]
It doesn't work either, but even if I put something in between the q-carousel tags it is not being displayed. However I tried using the same v-for statement with a div and the pictures are being displayed correctly, but I still don't know what is not making this carousel work

**Reply by MilosPaunovic** [reply]
Could you share a bit more of your code?

I've created this reproduction and it works as expected:
```js

<template>
  <q-layout view="lHh Lpr lFf">
    <q-carousel
      animated
      v-model="slide"
      arrows
      navigation
      infinite
    >
      <q-carousel-slide 
        v-for="(image, i) in pictures"
        :key="i"
        :name="i + 1" 
        :img-src="`${cloudFrontUrl}/${image}`"
      />
    </q-carousel>
   </q-layout>
</template>

<script>
import { defineComponent,ref } from 'vue'

export default defineComponent({
  name: 'MainLayout',
  setup () {
    return {
      slide: ref(1),
      cloudFrontUrl: ref('https://picsum.photos'),
      pictures: ['200', '300']
    }
  }
})
</script>
```

**Comment by FrenchGithubUser** [comment]
I used your v-for, key and name and now it works ! Thanks !

**Comment by MilosPaunovic** [comment]
> I used your v-for, key and name and now it works ! Thanks !

If that's the case, sounds like your `QCarouselSlide` `name` prop was being set to `picture.id`, and your `QCarousel` `v-model` is being set to `ref(1)`, so, there is no slide with name 1, hence, nothing is showed, hence, appears broken.
